frontend/util.js

```diff
@@ -0,0 +1,85 @@
+import { isEmail } from 'validator';
+
+export const removeBadValuesAndDuplicates = (array) => {
+    const uniqueSet = new Set(array.map(i => i.trim().toLowerCase()))
+    return [...uniqueSet].filter(i => i);
+}
+
+// form validation utility methods
+export const validateField = (formData, formErrors, field) => {
+    switch(field) {
+        case 'fullName':
+            validateName(formData, formErrors, field);
+            break;
+        case 'email':
+            validateEmail(formData, formErrors, field);
+            break;
+        case 'studentId':
+            validateNumber(formData, formErrors, field, 8);
+            break;
+        case 'phoneNumber':
+            validateNumber(formData, formErrors, field, 10);
+            break;
+        case 'designCentreSafety':
+        case 'whmis':
+        case 'machineShop':
+            validateBoolean(formData, formErrors, field);
+            break;
+        case 'termDescription':
+        case 'subteam':
+        case 'nextTermRole':
+        case 'nextTermActivity':
+        case 'program':
+        case 'termStatus':
+        case 'memberType':
+        case 'nextSchoolTerm':
+        case 'nextTermRole':
+        case 'nextTermActivity':
+            validateExists(formData, formErrors, field);
+            break;
+    }
+}
+
+export const clearErrorMessages = (formErrors) => {
+    for (const field of Object.keys(formErrors)) {
+        formErrors[field] = false;
+    }
+}
+
+export const isInvalidPhoneNumber = (number) => {
+    return !number.match(/^[0-9]*$/) || number.length > 10;
+}
+
+export const isInvalidStudentId = (number) => {
+    return !number.match(/^[0-9]*$/) || number.length > 8;
+}
```

Comment:
No, I can't do that because this is intended to prevent users from typing letters and typing past 8 digits long.
These methods are run when the form input changes, not when the user submits the form.

Reply:
I think this validation is sufficient for now.

Reply:
Ohh got it, thanks for explaining. Makes sense.